js/src/Ice/Endpoint.js

```diff
@@ -1,62 +1,29 @@
-//
-// Copyright (c) ZeroC, Inc. All rights reserved.
-//
-
-import { Ice as Ice_Version } from "./Version.js";
-const { EncodingVersion } = Ice_Version;
+// Copyright (c) ZeroC, Inc.
 
 /**
  *  Base class providing access to the endpoint details.
  **/
-export class EndpointInfo {
-    constructor(underlying = null, timeout = 0, compress = false) {
-        this.underlying = underlying;
-        this.timeout = timeout;
-        this.compress = compress;
-    }
-}
+export class EndpointInfo {}
```

Comment:
Isn't this changing the values used for default constructed instances?

for example with the new code `new EndpointInfo` have undefined value for `timeout` instead of `0`.

Reply:
Yes, except we actually always overwrite everything later, usually with `fillWithEndpointInfo`.

All this code is not very elegant, and I believe its structure is a consequence of the local Slice mapping. We should refactor in a follow-up PR: reintroduce internal constructors, simplify the structure (EndpointInfo should not hold onto Endpoint), and more.